KAN-145 fix: 회원탈퇴 및 사용자 상태 확인 버그 수정 및 안정화

dayaelee opened the pull request:
### 🔗 관련 이슈
KAN-145

### 📌 작업 내용

- 회원 탈퇴 api 호출 로직 추가
- 회원 탈퇴시 유저 정보 리셋 (닉네임, 이미지)
- 로그인 api처리후 리다이렉트를 위한 로직 추가 

### 🧑‍💻 작업 유형
- [x] 새로운 기능 추가
- [x] 버그 수정
- [x] 코드 리팩토링
- [ ] 문서 수정
- [ ] CI/CD 변경

### 🚨 주의 사항
배포시에 배포를 위한 코드 수정이 필요함
-httpOnly 설정 변경 필요
-리다이렉트 관련 파라미터 스토어 값 변경 필요 

### ✅ 체크 리스트
- [x] 코드 리뷰어가 이해할 수 있도록 주석 및 설명을 추가했습니다.
- [ ] 테스트를 추가하거나 업데이트 했습니다.
- [x] 이 변경 사항이 다른 기능에 영향을 미치는지 확인했습니다.

dayaelee committed:
KAN-145 feat: 계정 삭제 API 엔드포인트 추가

dayaelee committed:
KAN-145 feat: 리다이렉트를 위한 로직 추가, 회원 탈퇴 로직 추가

dayaelee committed:
KAN-145 feat: KAN-145 chore: JWT 쿠키 설정에서 테스트용 httpOnly 값을 수정

dayaelee committed:
KAN-145 chore: JWT 쿠키 설정에서 테스트용 httpOnly 값을 수정

dayaelee committed:
KAN-145 fix: 회원탈퇴 로직 삭제

dayaelee committed:
KAN-145 refactor: isLoggedIn 변수명 변경

dayaelee committed:
KAN-145 feat: 사용자 데이터 처리 로직 추가

konempty commented on src/main/java/com/spring/be/jwt/controller/AuthController.java:
```diff
@@ -47,4 +48,9 @@ public ResponseEntity<?> googleLogout(@RequestHeader("Cookie") String cookieHead
+    public ResponseEntity<?> deleteUser(@CookieValue("jwtToken") String jwtToken) {
+        return authService.deleteUser(jwtToken);
+    }
 }
```
<img width="101" alt="스크린샷 2024-12-20 오후 11 13 44" src="https://github.com/user-attachments/assets/24ea6027-5f4d-413b-9a0c-3dcbaac440f1" />

파일끝 개행 설정해주세요!
[참고](https://seongwon.dev/Git/20220303-%ED%8C%8C%EC%9D%BC%EC%9D%98_%EB%A7%88%EC%A7%80%EB%A7%89_%EA%B0%9C%ED%96%89/)

dayaelee replied:
아하 해당 부분 빠져있네요 확인후 수정했습니다!

konempty commented on src/main/java/com/spring/be/jwt/controller/GoogleAuthController.java:
```diff
@@ -45,15 +45,15 @@ public ResponseEntity<GoogleAuthResponseDto> handleGoogleToken(@RequestBody Map<
         // JWT 쿠키 설정
         ResponseCookie jwtCookie = ResponseCookie.from("jwtToken", responseDto.getJwtToken())
-                .httpOnly(true)
+                .httpOnly(false)// 테스트용 - false
```
테스트가 끝나면 롤백해야할것 같아요!
아니면 아직도 테스트 중인걸까요?

dayaelee replied:
네 맞습니다 아직 테스트중이에요 ㅠ 배포하실떄 따로 말해주시면 수정해드린다고 말씀드렸어요!

sall6550 commented on src/main/java/com/spring/be/jwt/controller/AuthController.java:
```diff
@@ -47,4 +48,9 @@ public ResponseEntity<?> googleLogout(@RequestHeader("Cookie") String cookieHead
         String token = CookieUtils.extractTokenFromCookie(cookieHeader, "jwtToken");
         return authService.performGoogleLogout(token);
     }
+
+    @GetMapping("/delete")
+    public ResponseEntity<?> deleteUser(@CookieValue("jwtToken") String jwtToken) {
+        return authService.deleteUser(jwtToken);
+    }
```
<img width="856" alt="image" src="https://github.com/user-attachments/assets/c7c31bc5-54fd-478b-8a9e-8fa2b3a09d1f" />
예전에 멘토님이 언급해주셨던건데 아직 < ? > 로 되어있는부분이 많더라고요 동적으로 타입이 바뀌는게아니면 지정해주는게 좋을거같습니다

dayaelee replied:
관련한 부분 수정하였습니다!

sall6550 commented on src/main/java/com/spring/be/jwt/service/AuthService.java:
```diff
@@ -258,4 +289,94 @@ public KakaoAuthResponseDto authenticateKakaoUser(String authorizationCode) {
         return generateToken(userId);
     }
+
+    public ResponseEntity<?> performKakaoUnlink(String token) {
```
서비스코드에서 ResponseEntity 가 쓰인거같은데 서비스코드에서 쓰는게아닌 컨트롤러에서 사용하셔야합니다 수정부탁드립니다.

dayaelee replied:
관련 부분 수정했습니다!

sall6550 commented on src/main/java/com/spring/be/jwt/service/AuthService.java:
```diff
@@ -258,4 +289,94 @@ public KakaoAuthResponseDto authenticateKakaoUser(String authorizationCode) {
+        }
+    }
+
+    public ResponseEntity<?> deleteUser(String jwtToken) {
```
서비스코드에서 ResponseEntity 가 쓰인거같은데 서비스코드에서 쓰는게아닌 컨트롤러에서 사용하셔야합니다 수정부탁드립니다.

dayaelee replied:
관련 부분 수정했습니다!

sall6550 commented on src/main/java/com/spring/be/jwt/service/AuthService.java:
```diff
@@ -258,4 +289,94 @@ public KakaoAuthResponseDto authenticateKakaoUser(String authorizationCode) {
+        String platform = user.getSocialPlatform();
+
+        // 플랫폼별 로그아웃 처리
+        ResponseEntity<?> logoutResponse;
```
수정부탁드립니다

dayaelee replied:
관련 부분 수정했습니다!

dayaelee committed:
KAN-145 feat: 컨트롤러에서 서비스 계층 반환 값 처리 로직 수정

- 서비스 계층에서 DTO 반환 방식에 맞추어 컨트롤러에서 ResponseEntity 생성 로직 추가

dayaelee committed:
KAN-145 feat: 서비스 계층에서 역할 분리를 위한 DTO 반환 로직 적용

- 서비스 계층에서 ResponseEntity 반환 방식을 DTO 반환으로 변경
- 컨트롤러에서 HTTP 상태 코드와 헤더를 관리할 수 있도록 리팩토링

dayaelee committed:
KAN-145 feat: 공통 응답 DTO 클래스 생성 및 적용

- 서비스 계층에서 로그아웃, 삭제와 같은 공통 작업 결과를 반환하기 위해 CommonResponseDto 추가

dayaelee committed:
KAN-145 feat: 로그아웃 전용 DTO 구조 개선

nuyeo reviewed (approved):
고생하셨습니다~
